libraries/AP_Scripting/applets/trisonica-mini.lua

```diff
@@ -0,0 +1,100 @@
+-- Script decodes and logs wind sensor data for Trisonica LI-550 mini
+-- https://www.licor.com/products/trisonica/LI-550-mini
+
+-- Parameters:
+-- SCR_ENABLE 1
+-- SERIAL5_PROTOCOL 28
+
+-- In SITL, you can enable serial ports to connect to the real device.
+-- https://ardupilot.org/dev/docs/learning-ardupilot-uarts-and-the-console.html#the-8-uarts
+-- ./Tools/autotest/sim_vehicle.py -v Plane --console --map -A "--serial5=uart:/dev/ttyUSB0" -D
+
+-- Remember to change baud to 230k in the sensor setup and enable the fields you want.
+-- Also, enable 10Hz output instead of the default 5Hz.
+
+-- Example data string (excluding quotes, including the carriage return line feed ending)
+-- "S  00.08 S2  00.07 D  245 DV  033 U  00.06 V  00.03 W  00.05 T  55889220.00 C  346.68 H  17.92 DP  03.68 P -099.70 AD  0.0000000 AX  -2913 AY  -3408 AZ -16600 PI  011.4 RO  009.8 MX   -619 MY    845 MZ    782 MD  337 TD  337"
+
+-- Log severities
+local MAV_SEVERITY = {EMERGENCY=0, ALERT=1, CRITICAL=2, ERROR=3, WARNING=4, NOTICE=5, INFO=6, DEBUG=7}
+
+-- find the serial first (0) scripting serial port instance
+local port = serial:find_serial(0)
+
+if not port then
+    gcs:send_text(MAV_SEVERITY.EMERGENCY, "No Scripting Serial Port")
+    return
```

Comment:
Personal choice: `error("no trisonica scripting port")` modulo message wording.

Reply:
```suggestion
    gcs:send_text(MAV_SEVERITY.EMERGENCY, "no trisonica scripting port")
    return
```

Reply:
You could also assert the `find_serial` for a even more annoying warning.

Reply:
To be clear I don't care too much about the message wording, I just think it would be simpler and easier to just call `error` or `assert` instead of sending an emergency message and returning.